### PR TITLE
AuthorsV2 Api

### PR DESCRIPTION
**Pull request by sandeepm96**

### CODE REVIEW

File: src/cognition.go [hunk 1]
```diff
@@ -149,6 +149,22 @@ func (cognition *Cognition) GetAuthor_v1(requestObject map[string]interface{}) D
...
 
+func (cognition *Cognition) GetAuthors_v2(requestObject map[string]interface{}) Data {
+    serviceApi := ServiceApi{}
+    var err error
+    data := Data{}
+    //authorId
+    if requestObject["type"].(string) == "top-rated" || requestObject["type"].(string) == "most-active" {
+        data.AuthorDataV2, err = serviceApi.GetAuthors_v2(requestObject)
+    } else {
+        log.Println("Error: Author Type Invalid")
```
Comment (roshan1988) [inline src/cognition.go]: Need to send an error response from here to client

File: src/cognition.go [hunk 2]
```diff
@@ -149,6 +149,22 @@ func (cognition *Cognition) GetAuthor_v1(requestObject map[string]interface{}) D
...
+    }
+    if err != nil {
+        log.Println("Error while calling service to get author list:", err)
```
Comment (roshan1988) [inline src/cognition.go]: Need to send an error response from here to client

File: src/service_api.go [hunk 3]
```diff
@@ -647,6 +647,45 @@ func (serviceApi *ServiceApi) GetAuthorByIDBatch_v2(requestObject map[string]int
...
+        return authorDataV2, err
+    } else {
+        defer response.Body.Close()
+    }
+    body, err := ioutil.ReadAll(response.Body)
+    if err != nil {
+        fmt.Println("Error : GetAuthors_v2 : ", err)
+        return authorDataV2, err
+    }
+    json.Unmarshal(body, &authorDataV2)
```
Comment (roshan1988) [inline src/service_api.go]: Unmarshall error can also occur here

File: src/translation/apis.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+func (tm *TranslationManager) GetTranslation(word string, language string) string {
+    return tm.translations[word][language]
+}
+
+func init() {
+    translationManager = &TranslationManager{}
+    err := translationManager.initialize()
+    if err != nil {
+        fmt.Println("Error while inititalizing translation manager:", err)
```
Comment (roshan1988) [inline src/translation/apis.go]: Should you make this fatal, so that the sevice startup itself fails ?

File: src/cognition.go (hunk 1 above)
Comment (sandeepm96) [reply src/cognition.go]: already handled as bad request in controller

File: src/cognition.go (hunk 2 above)
Comment (sandeepm96) [reply src/cognition.go]: done

File: src/service_api.go (hunk 3 above)
Comment (sandeepm96) [reply src/service_api.go]: done

**Pull request merged**